pettingzoo/utils/wrappers/terminate_illegal.py

```diff
@@ -42,7 +42,6 @@ def step(self, action: ActionType) -> None:
         if self._prev_obs is None:
             self.observe(self.agent_selection)
         if isinstance(self._prev_obs, dict):
-            assert self._prev_obs is not None
```

Comment:
Why’d you remove this assertion? Think it should check if it exists first before checking the value, but I guess the assertion will fail either way if not

Reply:
I removed it because it is redundant and will never be called if the value is None.
The line before confirms self._prev_obs is a dict. If it's a dict, it's not None, so the assertion can never fail.